specification/network/resource-manager/Microsoft.Network/stable/2025-03-01/publicIpAddress.json

```diff
@@ -438,6 +438,156 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/publicIPAddresses/{publicIpAddressName}/reserveCloudServicePublicIpAddress": {
+      "post": {
+        "tags": [
+          "PublicIPAddresses"
+        ],
+        "operationId": "PublicIPAddresses_ReserveCloudServicePublicIpAddress",
+        "description": "Reserves the specified Cloud Service Public IP by switching its allocation method to Static. If rollback is requested, reverts the allocation method to Dynamic.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "publicIpAddressName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the public IP address.",
+            "pattern": "^[A-Za-z0-9][A-Za-z0-9_\\.-]*[A-Za-z0-9_]$",
+            "minLength": 1,
+            "maxLength": 80
```

Comment:
Other APIs do not have this pattern. If there is an existing resource that does not fall under this pattern, they cannot use this API ?